### PR TITLE
Growable Pots cleanup

### PR DESCRIPTION
**Pull request by Ilysen**
#### Summary
```SUMMARY: None```

#### Purpose of change
The idea of Growable Pots is a good one, but the code is *very* unclean. It seems like it was made as a personal mod, which is fine! But, since it's in the mainline, and has no listed maintainer, I thought it should be brushed up to better standards.

#### Describe the solution
I went through and changed all of the improper json code; in most cases, I replaced all of the duplicated json with `copy-from`s for some new dummy items, and gave the ripe pots a `DISASSEMBLE` use_action.

I also removed the difficulty from disassembling ripe pots (because losing the plants you just spent four weeks growing because you fail to retrieve them is Silly), lowercased all of the names, added some flavor bits to the descriptions, and renamed the mod to Garden Pots, as well as changing its description to actually describe what it does.

#### Describe alternatives you've considered
Not renaming the mod. I opted to do this, as `Growable pots` doesn't really tell you much on its own and confused me enough to not bother trying it, especially given the strangely nondescriptive description. `Garden Pots` is a unique title that attracts attention and the description actually provides info on what it is.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
I think you should use `abstract` field for the dummy items.

File: data/mods/Growable_pots/items.json [hunk 1]
```diff
@@ -4,9 +4,8 @@
     "id": "gp_pot",
```
Comment (ZhilkinSerg) [inline data/mods/Growable_pots/items.json]:
```suggestion
    "abstract": "gp_pot",
```

File: data/mods/Growable_pots/items.json [hunk 2]
```diff
@@ -19,276 +18,181 @@
   },
   {
     "type": "GENERIC",
-    "id": "gp_tomato",
-    "name": "Garden pot (growing tomato)",
-    "name_plural": "Garden pot (growing tomatoes)",
-    "description": "This is a garden pot growing tomatoes.  When ripe you need to activate to prepare for harvest.",
+    "id": "gp_pot_growing",
```
Comment (ZhilkinSerg) [inline data/mods/Growable_pots/items.json]:
```suggestion
    "abstract": "gp_pot_growing",
```

File: data/mods/Growable_pots/items.json [hunk 3]
```diff
@@ -19,276 +18,181 @@
...
     "volume": 10,
     "bashing": 1,
-    "cutting": 0,
-    "to_hit": 0,
-    "category": "tools",
+    "category": "tools"
+  },
+  {
+    "type": "GENERIC",
+    "id": "gp_pot_grown",
```
Comment (ZhilkinSerg) [inline data/mods/Growable_pots/items.json]:
```suggestion
    "abstract": "gp_pot_grown",
```

**Comment by ZhilkinSerg** [comment]
Made "id->abstract" suggestions, otherwise looks good.

**Pull request merged**